aruco_ros/src/aruco_detector.cpp

```diff
@@ -232,62 +240,65 @@ class ArucoSimple
     unsigned int error_condition = 0;
     std::string error_message = "";
     static tf::TransformBroadcaster br;
+    bool hasTransformToReference;
 
     tf::Matrix3x3(transform.getRotation()).getRPY(roll, pitch, yaw);
 
+    // Get TF between camera and reference frame
+    tf::StampedTransform cameraToReference;
+    cameraToReference.setIdentity();
+
+    if ( reference_frame == camera_frame ) {
+      hasTransformToReference = true;
+    } else {
+      hasTransformToReference = getTransform(reference_frame, camera_frame, cameraToReference);
+    }
+
     // Test for erroneous conditions
     // Note that order is important for error message
     // All error codes will be forwarded, so this is important
     // only if error messages are overlayed on the camera image
     if (abs(remainder (roll - expected_roll, 2*M_PI)) > roll_tolerance){
-      error_message = arucoMsg.ANGLE_TOO_STEEP_MESSAGE;
-      error_condition |= arucoMsg.ANGLE_TOO_STEEP;
+      error_message = aruco_msgs::Marker::ANGLE_TOO_STEEP_MESSAGE;
+      error_condition |= aruco_msgs::Marker::ANGLE_TOO_STEEP;
     }
     if (abs(remainder (pitch - expected_pitch, 2*M_PI)) > pitch_tolerance){
-      error_message = arucoMsg.CODE_NOT_FLAT_MESSAGE;
-      error_condition |= arucoMsg.CODE_NOT_FLAT;
-    }
-    if (marker.getDistanceFromCamera() < min_distance){
-      error_message = arucoMsg.TOO_CLOSE_MESSAGE;
-      error_condition |= arucoMsg.TOO_CLOSE;
-    }
-    if (marker.getDistanceFromCamera() > max_distance){
-      error_message = arucoMsg.TOO_FAR_MESSAGE;
-      error_condition |= arucoMsg.TOO_FAR;
+      error_message = aruco_msgs::Marker::CODE_NOT_FLAT_MESSAGE;
+      error_condition |= aruco_msgs::Marker::CODE_NOT_FLAT;
     }
     if (abs(remainder(yaw - expected_yaw, 2*M_PI))  > yaw_tolerance){
-      error_message = arucoMsg.CODE_TWISTED_MESSAGE;
-      error_condition |= arucoMsg.CODE_TWISTED;
+      error_message = aruco_msgs::Marker::CODE_TWISTED_MESSAGE;
+      error_condition |= aruco_msgs::Marker::CODE_TWISTED;
     }
     if (abs(remainder(yaw - expected_yaw, 2*M_PI))  > M_PI/2.){
-      error_message = arucoMsg.CODE_UPSIDE_DOWN_MESSAGE;
-      error_condition |= arucoMsg.CODE_UPSIDE_DOWN;
+      error_message = aruco_msgs::Marker::CODE_UPSIDE_DOWN_MESSAGE;
+      error_condition |= aruco_msgs::Marker::CODE_UPSIDE_DOWN;
```

Comment:
the error messages are overwritten, but the conditions are stacked - is that intentional?

Reply:
Yea, now that you bring it up it doesn't make all that much sense to stack the conditions if we're just logging one message. This was just carried over from what they were doing previously, but I think handling one error condition makes things a little more clear